enumerable_methods.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    result = []
+    my_each { |i| result << i if yield(i) }
+    result
+  end
+
+  def my_any?
+    my_each { |i| true if yield(i) }
+  end
+
+  def my_all?
+    my_each { |i| true if yield(i) == true }
+  end
+
+  def my_none?
+    true unless block_given?
+    result = false
+    my_each { |i| true unless yield(i) }
+    result
+  end
+
+  def my_count
+    i = 0
+    my_each { i += 1 }
+    i
+  end
```

Comment:
**[REQUIRED]**
* Counts the number of items in enum that are equal to item if an argument is given.
* Counts the number of elements yielding a true value if a block is given.